sympy/polys/tests/test_polytools.py

```diff
@@ -3567,3 +3569,15 @@ def test_issue_20985():
     w, R = symbols('w R')
     poly = Poly(1.0 + I*w/R, w, 1/R)
     assert poly.degree() == S(1)
+
+
+def test_issue_22988():
+    e = S('(3/100 + 1234/10*x)')
+    p = Poly(e.subs({k: k.round(15) for k in e.atoms(Rational)}))
+    assert str(p) == "Poly(123.4*x + 0.03, x, domain='RR')"
+    assert str(p.terms()) == (
+        '[((1,), 123.400000000000000), ((0,), 0.030000000000000000)]')
+    assert filldedent(srepr(p)) == filldedent("""
+        Poly(Add(Mul(Float('123.399999999999999994', precision=63), 
+        Symbol('x')), Float('0.030000000000000000001', precision=60)), 
+        Symbol('x'))""")
```

Comment:
In master this gives
```
>>> str(p)
"Poly(123.4*x + 0.0299999999999999989, x, domain='RR')"
>>> str(p.terms())
'[((1,), 123.400000000000000), ((0,), 0.0299999999999999989)]'
>>> print(filldedent(srepr(p)))

Poly(Add(Mul(Float('123.399999999999999994', precision=63),
Symbol('x')), Float('0.0299999999999999988898', precision=63)),
Symbol('x'))
```

Note that the precision is no longer '63' for the lower precision 0.03, it is 60.

Reply:
> low precision float reduces the precision of everything else

Here is a comparison of master and this PR with the `e + x` example you gave:
```python
master

>>> e = N(sin(1) ** 2 + cos(1) ** 2 - 1)
>>> srepr(Poly(e + x))
"Poly(Add(Mul(Float('1.0', precision=7), Symbol('x')), Float('-3.782e-124', precision=7)), Symbol('x'))"
>>> ^Z

this pR

>>> e = N(sin(1) ** 2 + cos(1) ** 2 - 1)
>>> srepr(Poly(e + x))
"Poly(Mul(Float('1.0', precision=7), Symbol('x')), Symbol('x'))"
```

So this PR isn't breaking anything worse than it is.

Reply:
My hunch is that the conversion of non-floats (like the Integer coefficient on x) should use max(float prec, 53).